### PR TITLE
Preparations for crates.io publishing

### PR DESCRIPTION
**Pull request by chrysn**
This supersedes #98. In this new approach, all version related stuff is pulled together in the workspace, necessitating changes in only one file (rather than every Cargo.toml) during version bumps.

<del>This is based on #144; as GitHub still can't do staggered PRs, please start reviewing starting at "chore: pull version numbers...", and don't merge before #144.</del>

### CODE REVIEW

File: consts/Cargo.toml [hunk 1]
```diff
@@ -1,13 +1,10 @@
 [package]
 name = "edhoc-consts"
-version = "0.1.0"
+version.workspace = true
```
Comment (geonnave) [inline consts/Cargo.toml:3]: Ahh! I was wondering if there was an easy way to keep all version numbers in sync. Great!

File: Cargo.toml [hunk 2]
```diff
@@ -33,6 +33,27 @@ default-members = [
 # features without this, causing forced "std" in downstream crate.
 resolver = "2"
 
+[workspace.package]
+version = "0.4.0-alpha.1"
+repository = "https://github.com/openwsn-berkeley/edhoc-rs/"
+license = "BSD"
```
Comment (chrysn) [inline Cargo.toml:39]: Note that this is plainly moving data around from the other crates. If you want to have #152 resolved in here right away, I'm happy to update this, but would prefer explicit confirmation in that #152 that it is to be resolved toward which-ever license (in addition to one of you merging this, which'll need to happen anyway).

Comment (malishav) [reply Cargo.toml:39]: Confirming BSD-3-Clause is the right license.

File: crypto/edhoc-crypto-rustcrypto/Cargo.toml [hunk 3]
```diff
@@ -1,12 +1,13 @@
 [package]
 name = "edhoc-crypto-rustcrypto"
```
Comment (geonnave) [inline crypto/edhoc-crypto-rustcrypto/Cargo.toml:2]: @chrysn feel free to add yourself as author of this and the crypto-trait crates.

**Comment by geonnave** [comment]
This looks good to me. Anything missing?

File: crypto/edhoc-crypto-rustcrypto/Cargo.toml (hunk 3 above)
Comment (chrysn) [reply crypto/edhoc-crypto-rustcrypto/Cargo.toml:2]: Done (wasn't sure we're doing `authors = `, cargo doesn't set it by default any more); crypto-trait has been taken up into -consts.

**Comment by chrysn** [comment]
I think this is ready. The open question of when to publish at crates.io (and on which name -- I've privately contacted the author of the edhoc crate whether he still plans to maintain that) can be answered well after this PR.

**Pull request merged**